core/src/providers/cohere.rs

```diff
@@ -434,40 +417,33 @@ impl CohereEmbedder {
     async fn embed(&self, text: Vec<&str>, truncate: Truncate) -> Result<Embeddings> {
         assert!(self.api_key.is_some());
 
-        let https = HttpsConnector::new();
-        let cli = Client::builder().build::<_, hyper::Body>(https);
-
-        let req = Request::builder()
-            .method(Method::POST)
-            .uri(self.uri()?)
+        let res = reqwest::Client::new()
+            .post(self.uri()?.to_string())
```

Comment:
Got a bit lazy here, there's likely a way to impl `IntoUrl` without copying

Reply:
fine